src/linker/Linker/LinkerAttributesInformation.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using Mono.Cecil;
+
+namespace Mono.Linker
+{
+	struct LinkerAttributesInformation
+	{
+		Dictionary<Type, object> _linkerAttributes;
```

Comment:
I think we'll need support for multiple attributes on a member

Reply:
I don't think we need to add multiple attributes at this moment but I'd change the signature to be `Dictionary<Type, Attribute> _linkerAttributes;`

Reply:
I implemented support for multiple attributes for two reasons:
* It includes a check for multiple instances for cases where we expect only one - and issues a warning (this is good even for this PR, although there's no way in C# to get to that situation right now)
* Sven will need it soon anyway, and since I'm at it ...